### PR TITLE
#578 fixes None and replaces with string empty

### LINKED ISSUE #578: Fail to create many new users
Hey,
When I tried to create new users only by email an name, I got the following exception: 

```
zenpy.lib.exception.APIException: {"error": {"title": "Invalid attribute", "message": "You passed an invalid value for the users.0.id attribute. Invalid parameter: users.0.id must be an integer from api/v2/users/create_many"}}
```

My users creation code:
```
  user = User(name=name, email=email)
  usersToCreate.append(user)

zenpy_client.users.create(api_objects=usersToCreate)
```

When I debugged the issue until the part that makes the api call, I found that the json serialization created a new key of 'id' with the value None. This leads to zendesk api 400 response that the id must be an integer.

When I tried the call in "evaluate expression", after dropping the 'id' keys from the json object, is succeed.

This is more of an issue on Zendesk side, but it would be nice if you could drop the Id keys in serialization when they are None (relevant only for creation). 

Thanks in advanced.

### PR DESCRIPTION
**Pull request by cryptomail**
Fixes #578 and replaces None with string empty.

cc @krenzler

### CODE REVIEW

File: zenpy/lib/util.py [hunk 1]
```diff
@@ -114,6 +114,13 @@ def id_of(x):
 
     return outer
 
+def dict_clean(items):
+    result = {}
+    for key, value in items:
+        if value is None:
+            value = ''
+        result[key] = value
+    return result
```
Comment (ecoologic) [inline zenpy/lib/util.py:123]: I like that this has been extracted in util and we don't chew this data in the method itself.

How about renaming it to `dict_replace_none_with(items, replacement='')`?

A bit more specific, more suitable for reusable util functions?

Comment (ecoologic) [reply zenpy/lib/util.py:123]: Still Minor (your call, easy): How about?

```py
def dict_replace(items, old_value, new_value=''):
    result = {}
    for key, value in items:
        if value == old_value:
            value = new_value
        result[key] = value
    return result
```

A bit clearer than `clean`, more suitable for reusable util functions?

And some free unit tests from a well known AI:


```py
import unittest

class TestDictReplace(unittest.TestCase):

    def test_dict_replace(self):
        self.assertEqual(dict_replace([('key1', None), ('key2', 'value2')], None, 'default'), {'key1': 'default', 'key2': 'value2'})
        self.assertEqual(dict_replace([('key1', None), ('key2', None)], None, 'default'), {'key1': 'default', 'key2': 'default'})
        self.assertEqual(dict_replace([('key1', 'value1'), ('key2', 'value2')], None, 'default'), {'key1': 'value1', 'key2': 'value2'})
        self.assertEqual(dict_replace([('key1', None), ('key2', 'value2')], None), {'key1': '', 'key2': 'value2'})
        self.assertEqual(dict_replace([('key1', None), ('key2', None)], None), {'key1': '', 'key2': ''})
        self.assertEqual(dict_replace([('key1', 'value1'), ('key2', 'value2')], None), {'key1': 'value1', 'key2': 'value2'})

if __name__ == '__main__':
    unittest.main()
```

File: zenpy/lib/api.py [hunk 2]
```diff
@@ -282,7 +282,7 @@ def _serialize(self, zenpy_object):
             log.debug("Setting dirty object: {}".format(zenpy_object))
             self._dirty_object = zenpy_object
         return json.loads(
-            json.dumps(zenpy_object, default=json_encode_for_zendesk))
+            json.dumps(zenpy_object, default=json_encode_for_zendesk), object_pairs_hook=dict_clean)
```
Comment (cryptomail) [inline zenpy/lib/api.py:285]: @ecoologic what is the interface conformance for `object_pairs_hook`

Comment (cryptomail) [reply zenpy/lib/api.py:285]: https://docs.python.org/3/library/json.html passed in single dict

File: zenpy/lib/util.py (hunk 1 above)
Comment (cryptomail) [reply zenpy/lib/util.py:123]: https://github.com/facetoe/zenpy/pull/579/files#r1323802568 
Could build a layer of clean functions.
dict_clean
followed by the dict_clean_None?

**Pull request merged**